### PR TITLE
Add GetTransactionState syscall

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2579

### CODE REVIEW

File: src/neo/SmartContract/Native/LedgerContract.cs [hunk 1]
```diff
@@ -242,6 +242,14 @@ private Transaction GetTransactionForContract(ApplicationEngine engine, UInt256
             return state.Transaction;
         }
 
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates, Name = "getTransactionState")]
+        private VMState GetTransactionStateForContract(ApplicationEngine engine, UInt256 hash)
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/LedgerContract.cs:246]: `GetVMState`?

Comment (erikzhang) [reply src/neo/SmartContract/Native/LedgerContract.cs]: `getTransactionVMState` or `getVMStateForTransaction`?

Comment (shargon) [reply src/neo/SmartContract/Native/LedgerContract.cs]: I prefer `getTransactionVMState`

**Review by erikzhang — APPROVED** [review]

**Comment by erikzhang** [comment]
@shargon I'm afraid we have to revert this pr. We can't add method to a native contract now. Because we can't modify the abi or manifest of a native contract.

**Comment by roman-khimov** [comment]
Well, we have native contract versioning, maybe that's exactly the case where we can make some use of it.

**Pull request merged**